SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -360,8 +360,10 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
         
         // progressive decode the image in coder queue
         dispatch_async(self.coderQueue, ^{
-            // If all the data has already been downloaded, earily return to avoid further decoding
-            if (self.receivedSize >= self.expectedSize) {
+            // Get the finish status when progressive decoding task actually executes
+            BOOL finished = (self.receivedSize >= self.expectedSize);
+            // If at the beginning of decoding task was dispatched, the download was not finished. But by the time it executes, the download is finished, then earily return to avoid further decoding
+            if (!downloadFinished && finished) {
```

Comment:
We don't need do this, `downloadFinished` is always copy in block, that is, `!downloadFinished` is always true.

Reply:
`downloadFinished` is a scalar type but not reference by object. So if the URLSession delegate called first time all bytes was downloaded, the `downloadFinished` = YES. However, the coder queue may not be empty and contains the pending progressive decoding task with a partial image data. We don't need wait them to be finished. Earily cancel them, only let the final status (Current `receivedSize == expectedSize` && `downloadFinished` == YES) do progressive decoding. So that some coders can finish their final state. (Or this `updateIncrementalData:finished:` finished arg will always be NO)

Reply:
If this change is under consideration, maybe we can take further talk about it. I can just revert it to previous 4.x state.

Reply:
@dreampiggy I mean we just need to keep ` if (finished) ` 😂 .Let's assume `downloadFinished` is `YES`, when in block, `finished` is `YES`, right? so this condition, it would do image decode. It's what we want? I think we should also cancel it.

Reply:
@zhongwuzw This assume `(downloadFinished == YES) => (finished == YES)`. Current it's right. Maybe I just need think too much.

Reply:
@dreampiggy Emm, if `finished` == `YES`, we will directly decode in `- URLSession:task:didCompleteWithError:`, we got all image data right? maybe progressive decode is not unnecessary.

Reply:
@zhongwuzw Noop...Here is a short story.

Some of image coder, when implement progressive decoding, need to use many C level API, and they need a time to call API to stop progressive decoding and do a clean up. So there is an `finished` arg in progressive coder protocol. We'd better make it been called.

And, a ideal progressive decoder, should actually produce the same image when final image data is updated, as well as normal decoding with the same data. Like Apple's Image/IO for JPEG images. However, the real world is not. libwebp, for example, does not produce the same even we update the final full data, and it does not support progressive decoding Animated WebP. So we'd better to call normal decoding for final callback.